tests/test_views.py

```diff
@@ -170,6 +170,9 @@ def test_view_subset_shapes():
 
 
 def test_modify_view_component(matrix_type, mapping_name, request):
+    if "sparse_dask" in request.node.callspec.id:
+        msg = "sparse arrays in dask are generally expected to fail but in this case they do not"
+        pytest.xfail(msg)
```

Comment:
I’m very confused: `xfail` means:

> This is a feature that doesn’t work (yet or ever). We just ran test code, and detected that it failed, which is expected, as the feature doesn’t work.

Where does this code run a test and detect that it fails? If you want to *mark* it as a test that will fail, use `pytest.mark.xfail`. If you want to xfail it in a branch in the test code, use `pytest.xfail` in a branch.